### PR TITLE
[Auditbeat] Auditd: Check all fields are in fields.yml in unit test

### PR DESCRIPTION
**Pull request by cwurm**
Similar to how Python system tests in Metricbeat and Auditbeat call an `assert_fields_are_documented` function to check if all produced fields are in `fields.yml` this does the same in Go code for the `auditd` module where we don't have a Python system test.

I've built it while adding user fields in https://github.com/elastic/beats/pull/10456 and it was quite helpful to make sure I didn't miss anything.

The code is surprisingly simple and calls a lot of the "authoritative" code in Libbeat rather than try to replicate things like fields.yml parsing in Python. It makes me muse about moving the checks from Python to Go in other places as well.

### CODE REVIEW

File: auditbeat/module/auditd/audit_linux_test.go [hunk 1]
```diff
@@ -76,24 +78,54 @@ func TestData(t *testing.T) {
...
 
+	assertFieldsAreDocumented(t, events)
+
 	beatEvent := mbtest.StandardizeEvent(ms, events[0], core.AddDatasetToEvent)
 	mbtest.WriteEventToDataJSON(t, beatEvent, "")
 }
 
+// assertFieldsAreDocumented mimics assert_fields_are_documented in Python system tests.
+func assertFieldsAreDocumented(t *testing.T, events []mb.Event) {
+	fieldsYml, err := common.LoadFieldsYaml("../../fields.yml")
```
Comment (andrewkroh) [inline auditbeat/module/auditd/audit_linux_test.go]: Can you add `mg.Deps(Fields)` to the Go testing parts of the magefile. https://github.com/elastic/beats/blob/9bb6a94eed438971493256e377234dc2d9e9251c/auditbeat/magefile.go#L174-L188

Comment (cwurm) [reply auditbeat/module/auditd/audit_linux_test.go]: done

File: auditbeat/module/auditd/audit_linux_test.go [hunk 2]
```diff
@@ -76,24 +78,54 @@ func TestData(t *testing.T) {
...
+			for _, documentedFieldName := range documentedFields {
+				// Have to use HasPrefix and not "==" since fields in auditd.paths.* get flattened
+				// to auditd.paths which does not exist in fields.yml.
+				if strings.HasPrefix(documentedFieldName, eventFieldName) {
+					found = true
+				}
+			}
+			if !found {
+				assert.Fail(t, "Field not documented", "Key '%v' found in event is not documented.", eventFieldName)
+			}
```
Comment (webmat) [inline auditbeat/module/auditd/audit_linux_test.go]: Love the test!

However make sure not to use it with too many test events: this is O(n^3) 💥 

A quick improvement would be to cut the inner loop short, by breaking after setting `found = true`.

Comment (cwurm) [reply auditbeat/module/auditd/audit_linux_test.go]: Good point, let me add that `break`.

**Pull request merged**